node-notes-cli/app.js

```diff
@@ -0,0 +1,65 @@
+
+const fs = require('fs');
+const data = require('./data.json');
+const argument = process.argv[2];
+
+if (argument === 'read') {
+  readNote();
+  data.nextId = 5;
+} else if (argument === 'create') {
+  createNote();
+} else if (argument === 'delete') {
+  deleteNote();
+} else if (argument === 'update') {
+  updateNote();
+}
+
+function readNote() {
+  for (let i = 1; i < data.nextId; i++) {
+    if (data.notes[i]) {
+      console.log(i + ': ' + data.notes[i]);
+    }
+  }
+}
+
+function createNote() {
+  var noteArray = [];
+  for (let i = 3; i <= process.argv.length - 1; i++) {
+    if (process.argv[i]) {
+      noteArray.push(process.argv[i]);
+    }
+  }
+  var note = noteArray.join(' ');
+  data.notes[data.nextId] = note;
+  data.nextId = Object.keys(data.notes).length + 1;
```

Comment:
This doesn't work if you delete some of your notes. Just increment the nextId property.

```js
data.nextId++;
```